File: basis/src/main/java/com/Desafio/modelo/Funcionario.java
package com.Desafio.modelo;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import org.hibernate.validator.constraints.br.CPF;
import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;

@Getter
@Setter
@Entity
@Table(name = "funcionario")
@NoArgsConstructor
public class Funcionario implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int cod ;

    private String nome;

    private Date dataNascimento;

    @CPF
    private String cpf;

    @JoinColumn( name="cod_empresa", referencedColumnName = "cod")
    @ManyToOne
    private Empresa empresa;

    @JoinColumn( name = "cod_endereco", referencedColumnName = "cod")
    @OneToOne(cascade = CascadeType.ALL)
    private Endereco endereco;

}

File: desafio/src/app/Modelos/Funcionario.ts
export class Funcionario {
    cod:number;
    nome:String;
    dataNascimento: Date;
    cpf: String;
    codEmpresa:number;
   
    constructor(cod?:number, nome?:String,dataNascimento?: Date, cpf?: String,codEmpresa?:number){
        this.cod = cod,
        this.nome = nome,
        this.dataNascimento = dataNascimento,
        this.cpf = cpf,
        this.codEmpresa = codEmpresa;
    };
    
}